openhands/agenthub/codeact_agent/codeact_agent.py

```diff
@@ -104,14 +104,26 @@ def __init__(
             )
             self.prompt_manager = PromptManager(
                 microagent_dir=os.path.join(os.path.dirname(__file__), 'micro') if self.config.use_microagents else None,
+                custom_microagent_dir=os.path.join(
+                    workspace_base, '.openhands', 'microagents'
+                )
+                if workspace_base
+                else None,
```

Comment:
unfortunately while this probably works pretty well in a dev setup, it won't work very well in most setups :/

The right way to do this is to pull the microagents from the `runtime`, using the `list_files` function (you can see it [here](https://github.com/All-Hands-AI/OpenHands/blob/a93f1402debd325dac68360650bd12ae6abad643/openhands/server/listen.py#L491))

But the tricky thing is that now we need to pass data between the runtime and the agent controller, which don't typically talk to each other...

Reply:
I think what you'd want to do is insert a line here:

https://github.com/All-Hands-AI/OpenHands/blob/a93f1402debd325dac68360650bd12ae6abad643/openhands/server/session/agent_session.py#L113

you could pull the microagents from the runtime, then pass them to the AgentController

Reply:
Unfortunately there are several other places (main.go, cli.go, the github resolver which is an external project) that we'd need to do the same plumbing in. But I think we can start with this, and then refactor to get it plumbed in elsewhere